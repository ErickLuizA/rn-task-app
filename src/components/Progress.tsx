import * as React from 'react'
import { Dimensions, StyleSheet, View } from 'react-native'
import { useTheme } from 'react-native-paper'
import { Text } from 'react-native'
import { ITask } from '../context/TaskContext'

interface IProgressProps {
  progressType: string
  tasks: ITask[]
}

const width = Dimensions.get('screen').width

export default function Progress({ progressType, tasks }: IProgressProps) {
  const { colors } = useTheme()

  const totalTasks = tasks.length

  const doneTasks = tasks.filter(task => task.Done === true)

  // function getProgress(): number {
  //   const progress = (doneTasks.length * 100) / totalTasks

  //   if (doneTasks.length === 0 && totalTasks === 0) {
  //     return 0
  //   }

  //   return progress
  // }

  return (
    <View
      style={
        progressType === 'Daily'
          ? styles.container
          : styles.notDashboardContainer
      }
      testID="progressContainer">
      {/* <AnimatedCircularProgress
        size={120}
        width={15}
        fill={getProgress()}
        rotation={360}
        tintColor={colors.secondary}
        backgroundColor={colors.primary}>
        {fill => {
          return (
            <Text style={[styles.progressText, { color: colors.secondary }]}>
              {' '}
              {fill}%{' '}
            </Text>
          )
        }}
      </AnimatedCircularProgress> */}
      <View>
        <Text style={[styles.text, { color: colors.grayText }]}>
          {' '}
          {progressType} progress{' '}
        </Text>
        <Text style={[styles.taskText, { color: colors.grayText }]}>
          {' '}
          {doneTasks.length}/{totalTasks} tasks done{' '}
        </Text>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    width: width / 1.2,
    alignItems: 'center',
  },

  notDashboardContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    width: width / 1.2,
    alignItems: 'center',
    marginTop: 40,
  },

  progressText: {
    fontSize: 26,
    fontFamily: 'Roboto-MediumItalic',
  },

  text: {
    fontFamily: 'Roboto-Regular',
    fontSize: 20,
  },

  taskText: {
    fontSize: 18,
  },
})
