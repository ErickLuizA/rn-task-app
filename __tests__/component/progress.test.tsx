import * as React from 'react'
import { create } from 'react-test-renderer'
import Progress from '../../src/components/Progress'

describe('<Progress />', () => {
  it('render the Progress screen', () => {
    const tree = create(<Progress progressType="Daily" />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
