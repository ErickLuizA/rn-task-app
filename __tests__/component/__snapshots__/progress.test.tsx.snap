// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> render the Progress screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "width": 625,
    }
  }
  testID="progressContainer"
>
  <View>
    <RNSVGSvgView
      bbHeight={120}
      bbWidth={120}
      focusable={false}
      height={120}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 120,
            "width": 120,
          },
        ]
      }
      width={120}
    >
      <RNSVGGroup>
        <RNSVGGroup
          matrix={
            Array [
              1,
              0,
              -0,
              1,
              0,
              0,
            ]
          }
        >
          <RNSVGPath
            d="M 59.96701327930773 7.500010363084286 A 52.5 52.5 0 1 0 60 7.5"
            fill={0}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
              ]
            }
            stroke={4284612846}
            strokeLinecap={0}
            strokeWidth={15}
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 45,
          "height": 90,
          "justifyContent": "center",
          "left": 15,
          "overflow": "hidden",
          "position": "absolute",
          "top": 15,
          "width": 90,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-MediumItalic",
              "fontSize": 26,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
         
        0
        %
         
      </Text>
    </View>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto-Regular",
            "fontSize": 20,
          },
          Object {
            "color": "#000000",
          },
        ]
      }
    >
       
      Daily
       progress
       
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    >
       
      2
      /
      20
       tasks done
       
    </Text>
  </View>
</View>
`;
